nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -443,6 +487,58 @@ public static void checkTypeParameterNullnessForAssignability(
     }
   }
 
+  private @Nullable Type inferMethodTypeArgument(
+      Type typeParam, List<Type> lhsTypeArg, List<Type> typeArg, VisitorState state) {
+    Type inferType = null;
```

Comment:
Declare this variable inside the `else if` case in the loop.  Declaring it up here obscures the logic